### PR TITLE
expression: implement vectorized evaluation for ` builtinTruncateIntSig`

### PR DESCRIPTION
**Pull request by big747386**
### What problem does this PR solve? 
Implement vectorized evaluation for builtinTruncateIntSig #12103 

### What is changed and how it works?
```
BenchmarkVectorizedBuiltinMathFunc/builtinTruncateIntSig-VecBuiltinFunc-4                  	  100000            15632 ns/op	       0 B/op	       0 allocs/op
BenchmarkVectorizedBuiltinMathFunc/builtinTruncateIntSig-NonVecBuiltinFunc-4               	   20000            58218 ns/op	       0 B/op	       0 allocs/op**
```
### Check List 

Tests 

 - Unit test
 - Integration test

### CODE REVIEW

File: expression/builtin_math_vec.go [hunk 1]
```diff
@@ -697,11 +697,38 @@ func (b *builtinCeilIntToDecSig) vecEvalDecimal(input *chunk.Chunk, result *chun
...
+
+	if err := b.args[1].VecEvalInt(b.ctx, input, buf); err != nil {
+		return err
+	}
+	result.MergeNulls(buf)
+	i64s := result.Int64s()
+
+	for i := 0; i < len(i64s); i++ {
+		if result.IsNull(i) {
+			i64s[i] = 0
```
Comment (qw4990) [inline expression/builtin_math_vec.go]:
```suggestion
```

File: expression/builtin_math_vec.go [hunk 2]
```diff
@@ -697,11 +697,38 @@ func (b *builtinCeilIntToDecSig) vecEvalDecimal(input *chunk.Chunk, result *chun
...
+			continue
+		}
+		if buf.GetInt64(i) < 0 {
```
Comment (qw4990) [inline expression/builtin_math_vec.go]: Please use `buf.Int64s()` instead of using `buf.GetInt64(i)` and then we can reduce some function calls.

Comment (big747386) [reply expression/builtin_math_vec.go]: OK, I have fixed it

Comment (qw4990) [reply expression/builtin_math_vec.go]: Nice, and please also update the benchmark results.

File: expression/builtin_math_vec.go [hunk 3]
```diff
@@ -697,11 +697,39 @@ func (b *builtinCeilIntToDecSig) vecEvalDecimal(input *chunk.Chunk, result *chun
...
+	if err := b.args[1].VecEvalInt(b.ctx, input, buf); err != nil {
+		return err
+	}
+	result.MergeNulls(buf)
+	i64s := result.Int64s()
+	buf64s := buf.Int64s()
+
+	for i := 0; i < len(i64s); i++ {
+		if result.IsNull(i) {
+			i64s[i] = 0
```
Comment (Reminiscent) [inline expression/builtin_math_vec.go]: Maybe we can delete this line. Because when the ```result.IsNull(i)```, we won't visit i64[i]

Comment (big747386) [reply expression/builtin_math_vec.go]: oh, I saw the unvec function, it returns 0 if the result is null, so I think we should make it 0. All right, I will delete this line.

**Comment by Reminiscent** [comment]
Thanks! @big747386   Would you like to join our channel **#wg-vec-expr** in the [tidbcommunity](https://pingcap.com/tidbslack) slacks workspace for more discussion?

**Comment by big747386** [comment]
I have already joined the channel

**Pull request merged**